examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+      arguments: [],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: deployer, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log("Init LP Pool success. - ", response.hash);
+  return response.hash;
+};
+
+const createFungibleAsset = async (aptos: Aptos, admin: Account): Promise<void> => {
+  await new Promise<void>((resolve) => {
+    readline.question(
+      `Follow the steps to publish the Dog and Cat Coin module with Admin's address, and press enter. \n` +
+        "1. cd to /aptos-ts-sdk/examples/typescript/facoin folder \n" +
+        "2. run 'aptos move publish --named-address FACoin=[admin] --profile=[admin] \n" +
+        "   Note: [admin] is the same profile you used to publish your 'swap' package",
+      () => {
+        resolve();
+      },
+    );
+  });
+};
+
+/** Admin mint the coin*/
+const mintCoin = async (aptos: Aptos, admin: Account, amount: number | bigint, coinName: string): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: admin.accountAddress.toString(),
+    data: {
+      function: `0x${admin.accountAddress.toStringLongWithoutPrefix()}::${coinName}::mint`,
+      arguments: [admin.accountAddress, new U64(amount)],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: admin, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log(`Minting ${coinName} coin successful. - `, response.hash);
+  return response.hash;
+};
+
+const example = async () => {
+  console.log(
+    "This example will create a main user account called 'Admin', it will be used to deploy Liquidity pool and two new fungible assets. \n" +
+      "After creating the Dog and Cat coin, and the liquidity pool, it will swap one token for another. \n" +
+      "Note: This example requires you to have the 'swap' module published before running. \n" +
+      "If you haven't published the 'swap' module, please publish the package using \n" +
+      "'aptos move create-resource-account-and-publish-package --seed 0 --address-name=swap --named-addresses deployer=[admin] --profile [admin]' first. \n" +
+      "[admin] is the account profile you will be using for this example. \n",
+  );
+
+  // Prerequisite check
+  if (process.argv.length !== 4) {
+    console.log("Required usage: pnpm swap [swap_address] [user_private_key]");
+    process.exit(1);
+  }
+
+  const aptos = new Aptos();
+  // Create three accounts
+  const admin = Account.fromPrivateKey(new Ed25519PrivateKey(process.argv[3]));
+  const swapAddress = AccountAddress.fromHexInput(process.argv[2]);
+
+  console.log("====== Account info ======\n");
+  console.log(`Admin's address is: ${admin.accountAddress.toString()}`);
+  console.log(`Admin's private key is: ${admin.privateKey.toString()}`);
```

Comment:
Note, we should not print out private keys in examples.  Being as we should show proper best practices.
 
Instead, you can show the public key, or just the address.